build_membership.py

```diff
@@ -0,0 +1,225 @@
+import json
+
+
+def build_membership():
+    """Generate a .json file with all members of PySide2"""
+
+    # NOTE: PySide2, as of this writing, is incomplete.
+    # In it's __all__ module is a module, `QtOpenGL`
+    # that does no exists. This causes `import *` to fail.
+
+    from PySide2 import __all__
+    __all__.remove("QtOpenGL")
+
+    # These modules do not exist pre-Qt 5,
+    # so do not bother testing for them.
+    __all__.remove("QtSql")
+    __all__.remove("QtSvg")
+
+    from PySide2 import *
+
+    # Serialise members
+    members = {}
+    for name, module in locals().copy().items():
+        if name.startswith("_"):
+            continue
+
+        if name in ("json", "members"):
+            continue
+
+        members[name] = dir(module)
+
+    # Write to disk
+    with open("reference_members.json", "w") as f:
+        json.dump(members, f, indent=4)
+
+
+def build_tests():
+    """Build membership tests
+
+    Members only available in Qt 5 are excluded, along with member
+    exclusive to a paricular binding.
+
+    """
+
+    header = """\
+import os
+import sys
+import json
+
+with open("reference_members.json") as f:
+    reference_members = json.load(f)
+
+excluded = {excluded}
+
+""".format(excluded=json.dumps(excluded, indent=4))
+
+    test = """\
+def test_{binding}_members():
+    os.environ["QT_PREFERRED_BINDING"] = "{Binding}"
+
+    if "PyQt" in "{Binding}":
+        # PyQt4 and 5 performs some magic here
+        # that must take place before attempting
+        # to import with wildcard.
+        from Qt import Qt as _
+
+    if "PySide2" == "{Binding}":
+        # PySide2, as of this writing, doesn't include
+        # these modules in it's __all__ list; leaving
+        # the wildcard import below untrue.
+        from Qt import __all__
+        for missing in ("QtWidgets",
+                        "QtXml",
+                        "QtHelp",
+                        "QtPrintSupport"):
+            __all__.append(missing)
```

Comment:
> PySide2, as of this writing, doesn't include these modules in it's __all__ list; leaving the wildcard import below untrue.

Whoa. Isn't that a bug then?
I mean, QtWidgets should be included in `__all__`?